### DISCUSSION TITLE
Is there any way to compensate for input/judgement offset?

### DISCUSSION THREAD

**Discussion by ShouNichi** [ppy/osu#24495] [Q&A] [answered]
Hello everyone!
I'm having some problems with offsets.
No matter how I adjust the offsets (audio/beatmap), I would always be 21ms early (according to the result page).
By writing a small program like this
`if some_key.is_pressed(){
    time.sleep_ms(21);
    another_key.press()
}`
I can fix it partially, but not in an idiomatic way.
So I'm wondering if there is a way to adjust input offset?
I know it will cause hitsounds issue but I'm using my keyboard clicks as hitsounds so it's not a problem for me.
Any suggestions?
Thanks a lot.

**Comment by nagi-desuuu** [comment]
There might be something wrong with your hardware. Care to give us your specs?

**Reply by ShouNichi** [reply]
CPU: Intel I5-8400
GPU:CoffeeLake-S GT2 (UHD Graphics 630) / Nvidia GTX 1060(not used)
Mem: DDR4 2600 48G
Display: No idea... Some generic 60 fps integrated display?
OS: Arch Linux

I don't know what you need, so if you want any other info, feel free to ask!

**Comment by bdach** [comment]
Please read https://gist.github.com/smoogipoo/b2a9b95b4ea258345aaf599e1b47cbb5 for a primer on offsets.

**Reply by ShouNichi** [reply]
Thank you! It's a very nice read. However, I still don't know what causes the problem since I'm using my keyboard clicks as hitsounds so there should be basically no hitsound latency there.

When I turn off the sound entirely and only relying on visual, I'm still get 21-23 ms early anyway...?

And when I change the beatmap offset using the built-in "Audio settings - beatmap offset - calibrate using last play", I will still be 21 ms early and this process will be repeated recursively. So 21ms- another 21 ms (42 ms) - another 21 ms ... until I cannot adjust it anymore due to it's 50ms limit.

So it makes me wondering, maybe my skin has its judgement line wrongly aligned to the real judgement area?

Anyway, thanks again for your help and for maintaining this fantastic game.

**Reply by alexpado** [reply, marked as answer]
Have you tried any other skin ? 

I'm also using my keyboard as hitsounds, playing with -74ms offset due to wireless headset on top of that and I don't have the issue you are reporting, so it might just be the skin (I'm using one I made myself, so I can't really compare).

**Reply by ShouNichi** [reply]
Using argon fixed the problem, so it is indeed a problem of my skin.
Thanks! Maybe I'll try to modify it someday.
